docs/doc/90-contributing/02-pgo.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Profile Guided Optimization (PGO)
+sidebar_label: Profile Guided Optimization (PGO)
+description:
+  Building an optimized Databend with Profile Guided Optimization.
+---
+
+Profile-guided Optimization is a compiler optimization technique, which collects typical execution data (possible branches) during program execution and then optimizes for inlining, conditional branches, machine code layout, register allocation, etc.
+
+According to the tests, PGO makes a consistent improvement for Databend. The performance benefits depend on your typical workload - you can get better or worse results.
+
+More information about PGO in Databend you can read in [Issue #9387 | Feature: add PGO support](https://github.com/datafuselabs/databend/issues/9387).
+
+## Prerequisites
+
+Before you build Databend with PGO, make sure the following requirements have been met:
+
+- Install the PGO helper binary by adding the `llvm-tools-preview` component to your toolchain with rustup:
+
+```bash
+$ rustup component add llvm-tools-preview
+```
+
+- Install [`cargo-pgo`](https://crates.io/crates/cargo-pgo) that makes it easier to use [PGO](https://doc.rust-lang.org/rustc/profile-guided-optimization.html) and [BOLT](https://github.com/llvm/llvm-project/tree/main/bolt) to optimize Rust binaries.
```

Comment:
Since we haven't tested BOLT effects on Databend yet, I suggest removing BOLT from here. If you still want to mention BOLT in this guide, I suggest mentioning it somewhere at the end of the guide.

And BOLT is not exactly a PGO. Usually, it's called a PLO (Post Link Optimization). It's an important detail. The biggest reason to avoid recommending BOLT here just that we haven't tested it yet on Databend. Later, of course, we can expand the guide with BOLT as well.